### PR TITLE
Setup SSL with letsencrypt

### PR DESCRIPTION
**Pull request by tompluess**
Letsencrypt setup according to https://medium.com/@pentacent/nginx-and-lets-encrypt-with-docker-in-less-than-5-minutes-b4b8a60d3a71

And default ssl configuration of zammad, see https://github.com/zammad/zammad/blob/master/contrib/nginx/zammad_ssl.conf

### CODE REVIEW

File: init-letsencrypt.sh [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+if [ -d "$data_path" ]; then
+  read -p "Existing data found for $domains. Continue and replace existing certificate? (y/N) " decision
+  if [ "$decision" != "Y" ] && [ "$decision" != "y" ]; then
+    exit
+  fi
+fi
+
+
+if [ ! -e "$data_path/conf/options-ssl-nginx.conf" ] || [ ! -e "$data_path/conf/ssl-dhparams.pem" ]; then
```
Comment (crackofdusk) [inline init-letsencrypt.sh:19]: Is it safe to do that? In other projects, we hardcode the SSL options in the nginx configuration and generate Diffie-Hellman parameters locally.

**Review by crackofdusk — COMMENTED** [review]
It looks like `init-letsencrypt.sh` is an interactive program that needs to be run manually on server setup. In my opinion, we should consider automating that (not necessarily in this pull request), but it should at least be documented.

File: init-letsencrypt.sh (hunk 1 above)
Comment (tompluess) [reply init-letsencrypt.sh:19]: the tutorial i used proposes to use the recommended parameters by certbot. which i find a good idea.

the diffie-hellman parameters are not a secret. according to https://security.stackexchange.com/questions/94390/whats-the-purpose-of-dh-parameters

Comment (crackofdusk) [reply init-letsencrypt.sh:19]: Sorry, I did not state my concern correctly. Do we accept the following risks?

- GitHub being down or repository/files becoming somehow unavailable or compromised
- Not having the SSL options under version control

It's an acceptable risk for me, but I wanted to make sure it was considered.

Comment (tompluess) [reply init-letsencrypt.sh:19]: ok. lets keep the downloaded files under version control. 
but leave the init-script as is for documentation.

**Review by crackofdusk — APPROVED** [review]

**Pull request merged**